distributed/scheduler.py

```diff
@@ -758,12 +758,13 @@ def remove_worker(self, stream=None, address=None, safe=False):
             del self.worker_bytes[address]
 
             for key in self.has_what.pop(address):
-                self.who_has[key].remove(address)
-                if not self.who_has[key]:
-                    if key in self.tasks:
-                        recommendations[key] = 'released'
-                    else:
-                        recommendations[key] = 'forgotten'
+                if key in self.who_has:
```

Comment:
I would prefer to hunt down why these two were not matched.  I can try to find this sometime tomorrow.  What test was triggering this failure?  Generally I prefer to remain very strict in the scheduler.